se2_planning_ros/src/se2_planner_node.cpp

```diff
@@ -7,29 +7,76 @@
 
 #include <ros/ros.h>
 
+#include <grid_map_ros/grid_map_ros.hpp>
+#include "se2_planning/OmplReedsSheppPlanner.hpp"
+#include "se2_planning_ros/GridMapLazyStateValidatorRos.hpp"
 #include "se2_planning_ros/loaders.hpp"
 
-int main(int argc, char** argv) {
-  using namespace se2_planning;
+using namespace se2_planning;
+
+// Make planner global to have access to it in the gridMapCallback,
+// required to update planner bounds in dependence of grid map
+auto planner = std::make_shared<OmplReedsSheppPlanner>();
+
+void gridMapCallback(const grid_map_msgs::GridMap& msg) {
```

Comment:
This function here is solely to update the bounds in the planner, right? Since there is no other way to retrieve this from the state validator. Correct me if I am wrong

Reply:
Yes, exactly.